Refactoring and issue fixes

FoamyGuy opened the pull request:
In #163 it was suggested that further refactoring would be good in this library because `__init__.py` is rather long with everything used stuffed inside of it.

This PR is my attempt at that refactoring as well as fixing a few issues while I'm at it including #199 and #203, perhaps a few others.

So far I've broken up `__init__.py` into seperate files and fixed the imports enough to be able to run `circup --help` only, no testing with a device has been performed yet.

This commit: https://github.com/FoamyGuy/circup/commit/2572c32afe7ec3d8909362681b411a3718ded60f resolves #199 by allowing any command with `--help` to be run boardless.

Creating this as a draft because I'm sure the actions are not passing yet, and I intend to still come back and make further changes + carryout testing with a device. 

Any preliminary feedback is welcomed still.

tannewt commented on circup/class_definitions.py:
```diff
@@ -0,0 +1,345 @@
+CPY_VERSION = ""
+
+
+class Bundle:
```
I'd just split each of these classes into individual files of the same name. They'll be easier to find then.

FoamyGuy replied:
Bundle and Module are moved into their own respectively named files with the latest commits.

tannewt commented on circup/shared.py:
How is this different from util_functions?

FoamyGuy replied:
shared.py contains functions that are used by both the backends and the CLI command functions. 

util_functions.py (renamed to command_utils.py w/ latest commits) contains functions that are only used by the CLI command functions, not backends.

Granted It's kind of an odd distinction, they could be all moved into the same file if we wanted. I think the reason I made a new file instead of putting more into shared was to not contratict the comment in shared.py which states:
```
Utilities that are shared and used by both click CLI command functions
and Backend class functions.
```
But we could always tweak the wording if we do change the file anyway so I'm not sure why I felt the need to stick to it more strictly.

Let me know if you have a preference.

tannewt reviewed (commented):
Thanks for doing this refactor! I think adding file-level comments could help clarify what the purpose the contents share.

FoamyGuy commented:
Thanks for taking a look! I've added comments to each file with a brief description of contents. 

The latest commits also renamed `util_functions.py` to be a bit more descriptive of it's purpose (in my mind at least) `command_utils.py`

I've also done some very minimal testing now of `circup list` and `circup --verbose list` all testing via USB and haven't tried any other commands yet. 

I've started trying to remove the usage of global variables `CPY_VERSION` and `VERBOSE` since the places that reference them are in different files now those felt awkward. `CPY_VERSION` was fairly easy because it was used from a place that already had a backend which is where the value originated anyhow. 

`VERBOSE` was trickier, I was expecting to find some click API to access that but came up empty and resorted to checking for `--verbose` in `sys.argv` if there is a more proper way to do that I can change it

tannewt reviewed (commented):
Ah! Thank you! Looks good!

I'm not sure what the best way to structure a click cli app is. This is definitely an improvement.

FoamyGuy commented:
I think this is ready now. 

I had some trouble getting the tests to pass in the actions container. There seems to be some difference between that and my local environment because all were passing locally for me a while back but I needed to make some more changes to get them to pass in the actions container. 

I believe the only functional changes with this refactor should be the changes required to resolve : #199 and #203. 

I did carry out testing locally via USB and Web Workflow on a Feather S3 TFT with the commands: uninstall, install, list, bundle-add, bundle-remove, bundle-show, all appeared to be working as intended to me.

tannewt reviewed (approved):
Thank you!
